lib/analyzers/two_fer/analyze.rb

```diff
@@ -1,8 +1,275 @@
+      disapprove!(:no_method) unless main_method
+    end
+
+    def check_method_signature!
+      # If there is no parameter or it doesn't have a default value,
+      # then this solution won't pass the tests.
+      disapprove!(:missing_default_param) if paramaters.size != 1
+
+      # If they provide a splat, the tests can pass but we
+      # should suggest they use a real paramater
+      disapprove!(:splat_args, first_paramater_name) if first_paramater.restarg_type?
+
+      # If they don't provide an optional argument the tests will fail
+      disapprove!(:missing_default_param) unless first_paramater.optarg_type?
+    end
+
+    def check_for_optimal_solution!
+      # The optional solution looks like this:
+      #
+      # def self.two_fer(name="you")
+      #   "One for #{name}, one for me."
+      # end
+      # 
+      # The default argument must be 'you', and it must just be a single
+      # statement using interpolation. Other solutions might be approved
+      # but this is the only one that we would approve without comment.
+
+      return unless default_argument_is_optimal?
+      return unless one_line_solution?
+      return unless using_string_interpolation?
+
+      # If the interpolation has more than three components, then they've
+      # done something weird, so let's get a mentor to look at it!
+      refer_to_mentor! unless string_interpolation_has_three_components?
+
+      approve!
+    end
+
+    def check_for_correct_solution_without_string_interpolaton!
+      # If we don't have a correct default argugment or a one line
+      # solution then let's just get out of here.
+      return unless default_argument_is_optimal?
+      return unless one_line_solution?
+
+      loc = first_line_in_method(main_method)
+
+      # In the case of:
+      # "One for " + name + ", one for me."
+      if loc.method_name == :+ &&
+         loc.arguments[0].type == :str
+        approve!(:string_building)
+      end
+
+      # In the case of:
+      # format("One for %s, one for me.", name)
+      if loc.method_name == :format &&
+         loc.receiver == nil &&
+         loc.arguments[0].type == :str &&
+         loc.arguments[1].type == :lvar
+        approve!(:kernel_format)
+      end
+
+      # In the case of:
+      # "One for %s, one for me." % name
+      if loc.method_name == :% &&
+         loc.receiver.type == :str &&
+         loc.arguments[0].type == :lvar
+        approve!(:string_format)
+      end
+
+      # If we have a one-line method that passes the tests, then it's not
+      # soemthing we've planned for, so let's refer it to a mentor
+      return refer_to_mentor!
+    end
+
+    def check_for_conditional_on_default_argument!
+      loc = first_line_in_method(main_method)
+
+      # If we don't have a conditional, then let's get out of here.
+      #
+      # TODO: We might wnt to refactor this to extract a conditional from the
+      # method rather than insist on it being the first line.
```

Comment:
I think we probably do want to do that. On the other hand, if it's not the first line, does that mean that we want a mentor to look at it?